apps/src/music/player/soundSub.js

```diff
@@ -108,23 +112,38 @@ WebAudio.prototype.PlaySoundByBuffer = function (
   when,
   loop,
   effects,
-  callback
+  callback,
+  duration
 ) {
-  var source = audioContext.createBufferSource(); // creates a sound source
+  const source = audioContext.createBufferSource(); // creates a sound source
   source.buffer = audioBuffer; // tell the source which sound to play
+  let currentNode = source;
+
+  if (duration) {
+    // If playing for a specific duration, apply a small fadeout to the sound
+    // to prevent clicks and pops
```

Comment:
Is the click/pop the result of stopping a sound before it's reached the end of its sample?

Reply:
Yep exactly